client/src/components/users/UserProfile.tsx

```diff
@@ -0,0 +1,184 @@
+        const adminHeader = isAdmin ? <Header size="medium">User Administration</Header> : "";
+        const haveId = isAdmin ? <Popup inverted={true} trigger={<Form.Checkbox name="haveID"
+                                                                                label="Have ID"
+                                                                                checked={this.state.user.haveID}
+                                                                                onChange={this.handleInputChangeCheckbox}
+            />}
+                                        content="Whether the hardware desk currently has possession of this user's photo ID as collateral"/>
+            : "";
+        const userIsAdmin = isAdmin ? <Popup inverted={true} trigger={<Form.Checkbox label="Admin"
+                                                                                     name="admin"
+                                                                                     checked={this.state.user.admin}
+                                                                                     onChange={this.handleInputChangeCheckbox}
+            />}
+                                             content="Whether this user can access administration pages like this one"/>
+            : "";
+
+        const adminUuid = isAdmin ?
+            <Form.Field width={6}>
+                <label>UUID</label>
+                <p>{this.state.user.uuid}</p>
+            </Form.Field>
+            : "";
+
+        let yourThis = "your";
+        let pageTitle = "My Profile";
+        let aboutYou = "About You";
+
+        if (this.props.user && this.props.user.admin && this.props.user.uuid !== this.state.user.uuid) {
+            yourThis = "this";
+            pageTitle = `Edit ${this.state.user.name}`;
+            aboutYou = `About ${this.state.user.name}`;
+        }
+
+        return <div>
+            <Header size="huge">{pageTitle}</Header>
+            <Form loading={false}
+                  onChange={this.handleInputChange}
+                  error={!this.validateForm()}
+            >
+                <Header size="large">Contact Information</Header>
+                <Form.Field width={6}
+                            error={!this.state.user.slackUsername.length}
+                            required
+                >
+                    <label>Slack username</label>
+                    <Form.Input type="text"
+                                value={this.state.user.slackUsername}
+                                name="slackUsername"
+                                error={!this.state.user.slackUsername.length ? "This field is required" : null}
+                                required/>
+                </Form.Field>
+                <Form.Field width={6} required error={!this.validatePhone()}>
+                    <label>Phone number</label>
+                    <Cleave
+                        name="phone"
```

Comment:
what a peculiar package, is this the recommended solution for phone form inputs

Reply:
It matches the format Ground Truth expects, which I kinda disagree with so it might change the future